src/main/java/randoop/generation/Impurity.java

```diff
@@ -0,0 +1,518 @@
+   */
+  private static Sequence createSequence(
+      Sequence sequence, Executable executable, FuzzStatementOffset fuzzStatementOffset) {
+    Type outputType = determineOutputType(executable);
+    return createSequence(sequence, executable, outputType, fuzzStatementOffset, false);
+  }
+
+  /**
+   * Create a sequence for fuzzing an object of a given type using the given method. This overload
+   * allows output type and inclusion of explicit cast to be specified.
+   *
+   * @param sequence the sequence to append the fuzzing sequence to
+   * @param executable the method to be invoked to fuzz the object
+   * @param outputType the type of the object to be fuzzed
+   * @param fuzzStatementOffset the offset counter for the number of fuzzing statements added
+   * @param explicitCast whether to perform an explicit cast for the right-hand side of the fuzzing
+   *     statement
+   * @return a sequence with the fuzzing statement appended at the end
+   * @throws IllegalArgumentException if the method is not a method of the given type
+   */
+  private static Sequence createSequence(
+      Sequence sequence,
+      Executable executable,
+      Type outputType,
+      FuzzStatementOffset fuzzStatementOffset,
+      boolean explicitCast) {
+    CallableOperation callableOperation = createCallableOperation(executable, explicitCast);
+    NonParameterizedType declaringType = new NonParameterizedType(executable.getDeclaringClass());
+    List<Type> inputTypeList = getInputTypeList(executable, declaringType);
+    TypeTuple inputType = new TypeTuple(inputTypeList);
+    TypedOperation typedOperation =
+        new TypedClassOperation(callableOperation, declaringType, inputType, outputType);
+    List<Integer> inputIndex = calculateInputIndex(sequence, inputTypeList);
+    fuzzStatementOffset.increment(inputTypeList.size());
+    List<Sequence> sequenceList = Collections.singletonList(sequence);
+    return Sequence.createSequence(typedOperation, sequenceList, inputIndex);
+  }
+
+  /**
+   * Create a callable operation for fuzzing an object of a given type using the given method.
+   *
+   * @param executable the method to be invoked to fuzz the object
+   * @param explicitCast whether to perform an explicit cast for the right-hand side of the fuzzing
+   *     statement
+   * @return a callable operation for fuzzing an object of a given type using the given method
+   */
+  private static CallableOperation createCallableOperation(
+      Executable executable, boolean explicitCast) {
+    if (executable instanceof Method) {
+      return new MethodCall((Method) executable, explicitCast);
+    } else {
+      return new ConstructorCall((Constructor<?>) executable);
+    }
+  }
+
+  /**
+   * Determine the output type of the given method.
+   *
+   * @param executable the method to determine the output type of
+   * @return the output type of the given method
+   */
+  private static Type determineOutputType(Executable executable) {
```

Comment:
"determine" doesn't convey anything.  I would change the method name to `outputType()` or `getOutputType()`.